### PR TITLE
Import k6-docs eslint and prettier config

### PR DESCRIPTION
**Pull request by oleiade**
This PR imports the k6-docs eslint and prettier setup and applies it to the existing code.

### CODE REVIEW

**Review by na-- — CHANGES_REQUESTED** [review]
Hmm, the summary JS library is written specifically for k6, so there should be no need for Babel or any other transformations :thinking: In fact, it was written to be usable with `--compatibility-mode=base` from a long time ago, before many of the recent goja changes :sweat_smile: 

So it should be just a simple matter of bundling the JS files into a single one, even webpack might be an overkill. But I am not very familiar with the whole ecosystem, so someone else should probably review these parts.

:+1: :+1: for the linter and the syntax fixes, but I suggest that you should fist merge the [code changes from the k6 repo](https://github.com/grafana/k6/commits/master/js/summary.js) I mentioned in https://github.com/grafana/jslib.k6.io/issues/68#issuecomment-1263495062 and only then apply the style fixes.

**Comment by oleiade** [comment]
A lot of it is not necessary indeed, but I used the template so that one can generate a single build file from the two test files with the `npm run webpack` command. That way when publishing a new version, one runs that command and opens a PR with the generated `index.js` file.

It's a small improvement, but I believe it's a worthy one, and it also somewhat explains the process of publishing a jslib in my experience: it took me a while to actually understand how to proceed from scratch when I did it first.

Also, I have in mind to automate the jslib.k6.io repo in the future so that it either automatically detects new jslib releases from their repos, or have a process such as the one in the DefinetlyTyped repo. Either way, I believe having some build system (npm in that case) that produces a single output file would be beneficial with this still blurry objective in mind.  

In this PR, I would drop all the babel stuff, and only keep the `package.json` and webpack config. In another PR, I would update the code with the newer version that lives in the k6 repo as proposed, and would publish a new version as a result.

Let me know if that works for you 👍🏻 

PS: I'm still not 100% confident around the JS bundling/packaging/building ecosystem, please excuse any imprecisions on that front 🙇🏻

**Comment by na--** [comment]
> In this PR, I would drop all the babel stuff, and only keep the `package.json` and webpack config. In another PR, I would update the code with the newer version that lives in the k6 repo as proposed, and would publish a new version as a result.
> 
> Let me know if that works for you 👍🏻

:+1: works for me

> PS: I'm still not 100% confident around the JS bundling/packaging/building ecosystem, please excuse any imprecisions on that front 🙇🏻

I am probably less knowledgeable, so no worries  :sweat_smile:

**Comment by mstoykov** [comment]
> That way when publishing a new version, one runs that command and opens a PR with the generated index.js file.

But in this case we want *two* separate files:
1. to make the text summary
2. to make the junit report 

If somebody wants both - they can import both of them. 

Or have I misunderstood what you meant by that?

Arguably this is true for other dependencies (like the aws one-my original design was that the people wanting `s3` will import `s3` the people wanting `ssm` will import only it), but at least there - there is also additional work(compiling from typescript). Here there is nothing else from what I can see.

IMO we can just keep it as is and when/if those improvements to jslib happen we can change it here.

**Comment by oleiade** [comment]
@mstoykov What I instead (?) meant by that is that the current `index.js` file in the [jslib.k6.io repository ](https://github.com/grafana/jslib.k6.io/blob/main/lib/k6-summary/0.0.1/index.js), looks like the handstitched (concatenated) version of the two `text.js` and `junit.js` files. Therefore, I found the packaging useful to not have to do anything manually in the future, and to streamline under a single command the ability to open a PR to the jslib repo: `npm run webpack`, create a branch in the jslib repo, and put the result `index.js` file there (without said handmade steps).

Regarding the splitting into multiple destination files, I'm not against it, but considering the small size of the resulting file, I'm not sure splitting is worth it. In the AWS one where we could anticipate to have multiple services it did, but in this specific case, I wouldn't be so sure 🤓 

> IMO we can just keep it as is and when/if those improvements to jslib happen we can change it here.

I don't think I understood what you meant by that, are you suggesting I leave the repo as before the PR, or that I merge the PR as is ❓ 😄

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "k6-es6",
```
Comment (olegbespalov) [inline package.json:2]: Please don't forget to fill this (and other properties like `version`, `description` and so on) with a proper values

**Review by olegbespalov — COMMENTED** [review]
Not an expert in the area, but I share Ned's concern about the usage of webpack.

In general, it seems fine. Expect a small inline comment.

**Comment by mstoykov** [comment]
> looks like the handstitched (concatenated) version of the two text.js and junit.js files

This seems like an oversight that should be corrected not codified. 

> I don't think I understood

To keep it as is - no webpack/npm/babel and actually release v0.0.2 with the two files separated as part of jslib

**Comment by oleiade** [comment]
Thanks a lot for your input folks, much appreciated 🙇🏻 

As a summary, moving forward with this PR we will:
* remove the packaging from this PR
* keep the prettier/eslint config
* merge it and publish a v0.0.1 as a result

As a follow up, in a separate PR we will:
* Import the changes from k6
*  Create a version 0.0.2 (keeping a separate `text.js` and `junit.js` files
* Publish it to jslib.k6.io, exposing two separate files (thus, URLs too): `https://jslib.k6.io/summary/0.0.2/text.js` and `https://jslib.k6.io/summary/0.0.2/junit.js`

Let me know if I've misinterpreted something, or you wish to amend the steps 👍🏻

**Comment by oleiade** [comment]
I've applied the planned changes. Note that it appears the current version of the `text.js` file is up to date with the latest `summary.js` of the k6 repo (which only affects the text summary and not junit if I'm not mistaken).

**Review by olegbespalov — APPROVED** [review]
:+1:

**Comment by oleiade** [comment]
Ping @mstoykov @na-- for a 👍🏻

**Review by na-- — APPROVED** [review]

**Pull request merged**